Tests/Common/Client/src/GrpcChannelExt.cs

```diff
@@ -428,4 +467,81 @@ await foreach (var resp in streamingCall.ResponseStream.ReadAllAsync(cancellatio
+                                                                                     },
+                                                                },
+                                                      },
+                                             },
+                                             cancellationToken: cancellationToken)).Tasks.Single();
+
+    var end = await client.ListTasksAsync(new ListTasksRequest
+                                          {
+                                            Page     = 0,
+                                            PageSize = 1,
+                                            Filters = new Filters
+                                                      {
+                                                        Or =
+                                                        {
+                                                          FilterStatus(sessionId,
+                                                                       TaskStatus.Error,
+                                                                       TaskStatus.Completed,
+                                                                       TaskStatus.Retried),
+                                                        },
+                                                      },
+                                            Sort = new ListTasksRequest.Types.Sort
+                                                   {
+                                                     Direction = SortDirection.Desc,
+                                                     Field = new TaskField
+                                                             {
+                                                               TaskSummaryField = new TaskSummaryField
+                                                                                  {
+                                                                                    Field = TaskSummaryEnumField.EndedAt,
+                                                                                  },
+                                                             },
+                                                   },
+                                          },
+                                          cancellationToken: cancellationToken);
+
+    var last            = end.Tasks.Single();
+    var count           = end.Total;
+    var sessionDuration = (last.EndedAt - first.CreatedAt).ToTimeSpan();
+
+    logger.LogInformation("Throughput for session {session} : {sessionThroughput} task/s ({nTasks} tasks in {timespan})",
+                          sessionId,
+                          count / sessionDuration.TotalMilliseconds * 1000,
+                          count,
+                          sessionDuration);
+
+    return (long)(sessionDuration.TotalMilliseconds * 1000 / count);
```

Comment:
If you print the throughtput in this function, it can return void.

Reply:
I thought that if someone need to present  the data in another way, it would be good to return the info.

Reply:
Then, I would recommend to put the logs outside